src/components/ResultContainer.tsx

```diff
@@ -268,11 +268,16 @@ function ResultContainer({
     try {
       const subjectsResponse = await onDownload(buttonIndex, download);
       setDownloadError(null);
+      const timestamp = new Date()
+        .toISOString()
+        .replace('T', '-')
+        .replace(/:/g, '-')
+        .replace(/\..+/, '');
 
       const fileName =
         buttonIndex === 0
-          ? 'neurobagel-query-results.tsv'
-          : 'neurobagel-query-results-with-URIs.tsv';
+          ? `neurobagel-query-results-${timestamp}.tsv`
+          : `neurobagel-query-results-with-URIs-${timestamp}.tsv`;
```

Comment:
I suggest using `_` underscore here to better separate the Neurobagel section of the file name from the timestamp. Wdyt?

```suggestion
          ? `neurobagel-query-results_${timestamp}.tsv`
          : `neurobagel-query-results-with-URIs_${timestamp}.tsv`;
```

Reply:
Yes, we can do that!